cabal-install/Distribution/Client/CmdRepl.hs

```diff
@@ -13,34 +17,134 @@ module Distribution.Client.CmdRepl (
+         ( prettyShow )
+import Distribution.ReadE
+         ( ReadE, parsecToReadE )
+import qualified Distribution.SPDX.License as SPDX
+import Distribution.Solver.Types.SourcePackage
+         ( SourcePackage(..) )
+import qualified Distribution.Solver.Types.PackageIndex as PackageIndex
+import Distribution.Types.BuildInfo
+         ( BuildInfo(..), emptyBuildInfo )
 import Distribution.Types.ComponentName
-         ( componentNameString )
+         ( ComponentName(..), componentNameString )
+import Distribution.Types.CondTree
+         ( CondTree(..) )
+import Distribution.Types.Dependency
+         ( Dependency(..), thisPackageVersion )
+import Distribution.Types.GenericPackageDescription
+         ( emptyGenericPackageDescription )
+import Distribution.Types.PackageDescription
+         ( PackageDescription(..), emptyPackageDescription )
+import Distribution.Types.Library
+         ( Library(..), emptyLibrary )
+import Distribution.Types.PackageId
+         ( PackageIdentifier(..), PackageId )
+import Distribution.Types.UnqualComponentName
+         ( UnqualComponentName )
+import Distribution.Types.Version
+         ( mkVersion, version0, nullVersion )
+import Distribution.Types.VersionRange
+         ( anyVersion )
 import Distribution.Text
          ( display )
 import Distribution.Verbosity
-         ( Verbosity, normal )
+         ( Verbosity, normal, lessVerbose )
 import Distribution.Simple.Utils
-         ( wrapText, die', ordNub )
-
+         ( wrapText, die', ordNub, createTempDirectory, handleDoesNotExist )
+import Language.Haskell.Extension
+         ( Language(..) )
+
+import Control.Exception
+         ( catch, throwIO )
+import Control.Monad 
+         ( when, unless )
+import Data.List
+         ( sortOn )
 import qualified Data.Map as Map
+import Data.Ord
+        ( Down(..) )
 import qualified Data.Set as Set
-import Control.Monad (when)
-
-
-replCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags, [String])
+import System.Directory
+         ( getTemporaryDirectory, removeDirectoryRecursive )
+import System.FilePath
+         ( (</>) )
+
+type ReplFlags = [String]
+
+data EnvFlags = EnvFlags 
+  { envPackages :: [PackageId]
+  , envIncludeTransitive :: Flag Bool
+  , envOnlySpecified :: Flag Bool
+  }
+
+defaultEnvFlags :: EnvFlags
+defaultEnvFlags = EnvFlags
+  { envPackages = []
+  , envIncludeTransitive = toFlag True
+  , envOnlySpecified = toFlag False
+  }
+
+envOptions :: ShowOrParseArgs -> [OptionField EnvFlags]
+envOptions _ =
+  [ option ['p'] ["package"]
+    "Include an additional package in the environment presented to GHCi."
+    envPackages (\p flags -> flags { envPackages = p ++ envPackages flags })
+    (reqArg "PACKAGE" packageIdReadE (fmap prettyShow :: [PackageId] -> [String]))
+  , option [] ["no-transitive-deps"]
+    "Don't automatically include transitive dependencies of requested packages."
+    envIncludeTransitive (\p flags -> flags { envIncludeTransitive = p })
```

Comment:
I don't see how `envIncludeTransitive` is used.

Reply:
Because I haven't added code to include the transitive dependencies yet. :)